Client/Application/AppDelegate.swift

```diff
@@ -143,38 +123,36 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
         pushNotificationSetup()
 
-        if let profile = self.profile {
-            let persistedCurrentVersion = InstallType.persistedCurrentVersion()
-            let introScreen = profile.prefs.intForKey(PrefsKeys.IntroSeen)
-            // upgrade install - Intro screen shown & persisted current version does not match
-            if introScreen != nil && persistedCurrentVersion != AppInfo.appVersion {
-                InstallType.set(type: .upgrade)
-                InstallType.updateCurrentVersion(version: AppInfo.appVersion)
-            }
+        let persistedCurrentVersion = InstallType.persistedCurrentVersion()
+        let introScreen = profile.prefs.intForKey(PrefsKeys.IntroSeen)
+        // upgrade install - Intro screen shown & persisted current version does not match
+        if introScreen != nil && persistedCurrentVersion != AppInfo.appVersion {
+            InstallType.set(type: .upgrade)
+            InstallType.updateCurrentVersion(version: AppInfo.appVersion)
+        }
 
-            // We need to check if the app is a clean install to use for
-            // preventing the What's New URL from appearing.
-            if introScreen == nil {
-                // fresh install - Intro screen not yet shown
-                InstallType.set(type: .fresh)
-                InstallType.updateCurrentVersion(version: AppInfo.appVersion)
-                // Profile setup
-                profile.prefs.setString(AppInfo.appVersion, forKey: LatestAppVersionProfileKey)
-
-            } else if profile.prefs.boolForKey(PrefsKeys.KeySecondRun) == nil {
-                profile.prefs.setBool(true, forKey: PrefsKeys.KeySecondRun)
-            }
+        // We need to check if the app is a clean install to use for
+        // preventing the What's New URL from appearing.
+        if introScreen == nil {
+            // fresh install - Intro screen not yet shown
+            InstallType.set(type: .fresh)
+            InstallType.updateCurrentVersion(version: AppInfo.appVersion)
+            // Profile setup
+            profile.prefs.setString(AppInfo.appVersion, forKey: LatestAppVersionProfileKey)
+
+        } else if profile.prefs.boolForKey(PrefsKeys.KeySecondRun) == nil {
+            profile.prefs.setBool(true, forKey: PrefsKeys.KeySecondRun)
```

Comment:
nit: Considering we are on a cleaning spree, I'd suggest cleaning this up into its own area so that `didFinishLaunchingWithOptions` seems light

Reply:
It's on the list, I still have one or two more PR's around cleaning up this file for this ticket. I wanted to keep this one limited to just the profile change so it's hopefully easier to see any issues I might have over looked.